### PR TITLE
LF-4611 Add no animal locations modal

### PR DESCRIPTION
**Pull request by Duncan-Brain**
**Description**

Adds a modal that prevents movement task creation without animal locations.

Jira link:[ LF-4611](https://lite-farm.atlassian.net/browse/LF-4611)

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by kathyavini** [comment]
Where is the modal idea coming from? The solution we discussed with Loïc for this ticket was allowing more animal locations!

**Comment by Duncan-Brain** [comment]
@kathyavini From the linked ticket -- fences and sensors are still not allowed

I could add your residence in this pr if you want?

File: packages/webapp/public/locales/en/translation.json [hunk 1]
```diff
@@ -208,7 +208,9 @@
       "OTHER_PURPOSE_EXPLANATION_LABEL": "Explanation",
       "OTHER_PURPOSE_EXPLANATION_PLACEHOLDER": "Please describe the reason for this movement"
     },
+    "NEED_ANIMAL_LOCATION": "You'll need a location or area where animals can be moved to in order to move animals. Go to the map to create an animal location.",
```
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:211]:
```suggestion
    "NEED_ANIMAL_LOCATION": "You'll need a location or area where animals can be moved. Go to the map to create an animal location.",
```

**Comment by SayakaOno** [comment]
PR for updating translation key: https://github.com/LiteFarmOrg/LiteFarm/pull/3609

**Review by kathyavini — APPROVED** [review]

**Review by antsgar — APPROVED** [review]

**Pull request merged**